token/program-2022/src/processor.rs

```diff
@@ -947,6 +947,16 @@ impl Processor {
         }
         Ok(())
     }
+
+    fn get_required_account_extensions(
+        mint_account_info: &AccountInfo,
```

Comment:
nit: how about this takes in `StateWithExtensions::<Mint>`, and use the unpack done in line 115? that way we avoid doing two unpacks

Reply:
How about I instead remove this check:
https://github.com/solana-labs/solana-program-library/blob/3e1dd733968a74c19b7df5b0eda1663da79a2377/token/program-2022/src/processor.rs#L114-L116
and map the error type on the unpack in get_required_account_extensions to TokenError::InvalidMint?

That would reduce to one unpack for now. We could reassess if there are other methods that end up calling `get_required_account_extensions` and have other mint needs.

Reply:
works for me!